woocommerce/woocommerce
# Enhancement: Changing shopper feedback on payment response failure

### Changes proposed in this Pull Request:





Currently, if a payment gateway response fails or server chokes while processing an order payment, the shopper receives erroneous feedback suggesting to retry the order or technical error data. The order may already be paid for and present in the system.
This PR improves the feedback given, prompting the user to confirm the order and paymens instead of placing the order again.

Closes #41527 .

| Before | After|
| - | - |
| <img width="300" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/857c0cc5-ca6e-47d4-bf44-ec484b762c83"> | <img width="300" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/4ea467ba-f04d-4b0b-9c46-aa79c8048ea4"> |
| <img width="300" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/8a0fd12c-53e2-46c1-9b8c-73082e06a4eb"><br>_(example of technical error message)_ | <img width="300" alt="image" src="https://github.com/woocommerce/woocommerce/assets/17236129/6f68032c-8f5b-4596-b848-9770c680f9db"> |


### How to test the changes in this Pull Request:

1. Install the Code Snippets extension and add a new snippet with the following code:
```php
add_action("woocommerce_checkout_order_processed","rip",20,1);
function rip(){
	die('rip');
}
```
2. Place an order on **shortcode checkout**
3. Verify the above-expected msg is displayed, and the order history link works correctly
4. Go to the order history link , and verify the order was created and is `Pending Payment`
5. Deactivate the snippet, and verify you can place orders normally
6. Create and activate a new snippet with the following code:
```php
add_filter("woocommerce_payment_successful_result","rip",20,1);
function rip(){
	die('rip again');
}
```
7.  Repeat steps 2, 3
8. Go to the order history link and verify that the order was created and is `Processing`
9. Repeat step 5




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Changed shopper feedback on payment response failure

</details>
